aspnetcore/security/authentication/facebook-logins.md

```diff
@@ -0,0 +1,122 @@
+
+  ````
+  dotnet user-secrets set Authentication:Facebook:AppId <app-Id>
+     ````
+
+* Set the Facebook AppSecret
+
+  <!-- literal_block {"ids": [], "xml:space": "preserve"} -->
+
+  ````
+  dotnet user-secrets set Authentication:Facebook:AppSecret <app-secret>
+     ````
+
+The following code reads the configuration values stored by the [Secret Manager](../app-secrets.md#security-app-secrets):
+
+[!code-csharp[Main](../../common/samples/WebApplication1/Startup.cs?highlight=11&range=20-36)]
+
+## Enable Facebook middleware
+
+> [!NOTE]
+> You will need to use NuGet to install the [Microsoft.AspNetCore.Authentication.Facebook](https://www.nuget.org/packages/Microsoft.AspNetCore.Authentication.Facebook/1.1.0-preview1-final) package if it hasn't already been installed. Alternatively, execute the following in your project directory:
+>
+> `dotnet install Microsoft.AspNetCore.Authentication.Facebook`
+
+Add the Facebook middleware in the `Configure` method in `Startup.cs`:
+
+[!code-csharp[Main](./sociallogins/sample/Startup.cs?highlight=21,22,23,24,25&range=58-110)]
+
+## Login with Facebook
+
+Run your application and click Login. You will see an option for Facebook.
+
+![image](sociallogins/_static/FBLogin1.PNG)
+
+When you click on Facebook, you will be redirected to Facebook for authentication.
+
+![image](sociallogins/_static/FBLogin2.PNG)
+
+Once you enter your Facebook credentials, then you will be redirected back to the Web site where you can set your email.
+
+You are now logged in using your Facebook credentials:
+
+![image](sociallogins/_static/Done.PNG)
+
+## Optionally set password
+
+When you register with an external login provider, you do not have a password registered with the app. This alleviates you from creating and remembering a password for the site, but it also makes you dependent on the external login provider. If the external login provider is unavailable, you won't be able to log in to the web site.
+
+To create a password and login using your email that you set during the login process with external providers:
+
+* Tap the **Hello <email alias>** link at the top right corner to navigate to the **Manage** view.
+
+![image](sociallogins/_static/pass1.PNG)
+
+* Tap **Create**
+
+![image](sociallogins/_static/pass2.PNG)
+
+* Set a valid password and you can use this to login with your email
+
+## Next steps
+
+* This article showed how you can authenticate with Facebook. You can follow a similar approach to authenticate with [Microsoft Account](microsoft-logins.md), [Twitter](twitter-logins.md), [Google](google-logins.md) and other providers.
+
+* Once you publish your Web site to Azure Web App, you should reset the *AppSecret* in the Facebook developer portal.
+
+* Set the *Authentication:Facebook:AppId* and Authentication:Facebook:AppSecret as application setting in the Azure Web App portal. The configuration system is setup to read keys from environment variables.
```

Comment:
Code highlight instead of emphasis would work better, and apply it to both the `AppId` and the `AppSecret` labels